web-api/workflow-terraform/migration/main/lambdas/migration.ts

```diff
@@ -0,0 +1,114 @@
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
+import { DynamoDBDocument, PutCommandOutput } from '@aws-sdk/lib-dynamodb';
+import { createApplicationContext } from '@web-api/applicationContext';
+import { createLogger } from '@web-api/createLogger';
+import { migrateRecords as migrations } from './migration-segments';
+import type { DynamoDBStreamEvent, Handler } from 'aws-lambda';
+
+type migrationsCallback = {
+  (
+    applicationContext: IApplicationContext,
+    {
+      documentClient,
+      items,
+      ranMigrations,
+    }: {
+      documentClient: DynamoDBDocument;
+      items: Record<string, any>[];
+      ranMigrations?: { [key: string]: boolean };
+    },
+  ): Promise<any>;
+};
+
+const dynamodb = new DynamoDBClient({
+  maxAttempts: 10,
+  region: 'us-east-1',
+});
+
+const docClient = DynamoDBDocument.from(dynamodb, {
+  marshallOptions: { removeUndefinedValues: true },
+});
+
+export const processItems = async (
+  applicationContext: IApplicationContext,
+  {
+    documentClient,
+    items,
+    migrateRecords,
+  }: {
+    documentClient: DynamoDBDocument;
+    items: Record<string, any>[];
+    migrateRecords: migrationsCallback;
+  },
+): Promise<void> => {
+  const promises: Promise<PutCommandOutput>[] = [];
+
+  items = await migrateRecords(applicationContext, { documentClient, items });
+
+  for (const item of items) {
+    promises.push(
+      documentClient.put({
```

Comment:
If this is for the migration lambda I imagine the performance is quite important. If we are trying to do many puts or many deletes I think it is important that we use batchWrite() instead of individual puts or individual deletes. 

There are a couple of gotchas with batchWrite() like it can only handle 25 commands at a time, but let me know if I can provide any help on how to implement or point to examples.